.github/workflows/tests.yaml

```diff
@@ -57,11 +64,32 @@ jobs:
               timeout-minutes: 5
               run: |
                   scripts/examples/gn_build_example.sh examples/chip-tool out/debug/standalone/
+            - name: Copy objdir
+              run: |
+                  # The idea is to not upload our objdir unless builds have
+                  # actually succeeded, because that just wastes space.
+                  rsync -a out/debug/standalone/ objdir-clone || true
```

Comment:
What's wrong with just uploading the binaries in their original paths later in upload-artifact@v2 ?

Reply:
@mspang  Ideally we would only upload on crash.

We don't have a good way to detect "crash" here, so we are uploading on failure.

But if we always upload on failure, we will upload also in the case when the failure is a build failure.  And there's no reason to do that in that case.  I tried to explain that in the comment, but would welcome suggestions on making it clearer!

Reply:
I was mainly interested in why we need to rsync instead of passing out/debug/standalone/ to the upload task directly.

Reply:
Because then it would upload it even if the stuff in there has nothing to do with anything, basically.  This way we rsync only _after_ build is done, so only have things to upload if the build succeeded.

Reply:
Still not following. What's wrong with

```
            - name: Uploading objdir for debugging
              uses: actions/upload-artifact@v2
              if: ${{ failure() }}
              with:
                  name: crash-objdir-linux
                  path: out/debug/standalone/
                  # objdirs are big; don't hold on to them too long.
                  retention-days: 5
```

Reply:
If the build fails partway through, that would upload the objdir, with whatever object files have been built so far, right?

But if the build fails partway through we don't want to upload anything at all.